src/main/java/com/flippingutilities/utilities/GeHistoryTabExtractor.java

```diff
@@ -33,33 +36,47 @@ public static OfferEvent createOfferEventFromWidgetGroup(List<Widget> widgets) {
         GrandExchangeOfferState offerState = getState(widgets.get(2));
         int quantity = widgets.get(4).getItemQuantity();
         int itemId = widgets.get(4).getItemId();
-        int price = getPrice(widgets.get(5));
+        int price = getPrice(widgets.get(5), quantity);
         boolean isBuy = offerState == GrandExchangeOfferState.BOUGHT;
         Instant time = Instant.now();
         int totalQuantity = quantity;
         int tickArrivedAt = -1;
         //just making ticksSinceFirst offer something > 2 so it doesn't count as a margin check
         int ticksSinceFirstOffer = 10;
 
-        OfferEvent offerEvent = new OfferEvent(isBuy, itemId, quantity, price, time, slot, offerState, tickArrivedAt, ticksSinceFirstOffer, totalQuantity, true, null, null, false, null,0,0);
+        OfferEvent offerEvent = new OfferEvent(UUID.randomUUID().toString(), isBuy, itemId, quantity, price, time, slot, offerState, tickArrivedAt, ticksSinceFirstOffer, totalQuantity, true, null, null, false, null,0,0);
         return offerEvent;
     }
 
-    private static int getPrice(Widget w) {
+    private static int getPrice(Widget w, int quantity) {
         String text = w.getText();
         String numString = text;
-        if (text.contains("each")) {
-            Matcher m = PRICE_PATTERN.matcher(text);
-            m.find();
-            numString = m.group(1);
+        Matcher m;
+        boolean isTotalPrice = false;
+        if (text.contains(")</col>")) {
+            m = ORIGINAL_PRICE_PATTERN.matcher(text);
+            isTotalPrice = true;
         }
+        else if (text.contains("each")) {
+            m = MULTI_ITEM_PATTERN.matcher(text);
+        }
+        else {
+            m = SINGLE_ITEM_PATTERN.matcher(text);
+        }
+        m.find();
+        numString = m.group(1);
         StringBuilder s = new StringBuilder();
         for (char c : numString.toCharArray()) {
             if (Character.isDigit(c)) {
                 s.append(c);
             }
         }
-        return Integer.parseInt(s.toString());
+
+        int price = Integer.parseInt(s.toString());
+        if (isTotalPrice) {
+            return price/quantity;
+        }
```

Comment:
Description on why this can happen (GE history?)

Also, this is a bit lossy: It'll round down the int it has, which can slightly alter the computation of bulk trades
I'd document this on the method